src/components/Label/Label.tsx

```diff
@@ -1,21 +1,40 @@
 import React, { useMemo } from "react";
-import PropTypes from "prop-types";
 import cx from "classnames";
 import { backwardCompatibilityForProperties } from "../../helpers/backwardCompatibilityForProperties";
-import { LABEL_COLORS, LABEL_TYPES } from "./LabelConstants";
 import Leg from "./Leg";
 import "./Label.scss";
+import VibeComponentProps from "src/types/VibeComponentProps";
+
+const LABEL_TYPES = {
+  FILL: "fill",
+  LINE: "line"
+} as const;
+
+const LABEL_COLORS = {
+  PRIMARY: "primary",
+  DARK: "dark",
+  NEGATIVE: "negative",
+  POSITIVE: "positive"
+} as const;
+
+interface LabelProps extends VibeComponentProps {
+  wrapperClassName: string;
+  kind: typeof LABEL_TYPES[keyof typeof LABEL_TYPES];
+  color: typeof LABEL_COLORS[keyof typeof LABEL_COLORS];
+  text: string;
+  isAnimationDisabled: boolean;
+  isLegIncluded: boolean;
+}
 
 const Label = ({
-  // Backward compatibility for enum naming
   className,
   wrapperClassName,
-  kind,
-  color,
+  kind = "fill",
+  color = "primary",
   text = "",
-  isAnimationDisabled,
-  isLegIncluded
-}) => {
+  isAnimationDisabled = false,
+  isLegIncluded = false
+}: Partial<LabelProps>) => {
```

Comment:
I think it's best to mark the properties of `LabelProps` as optionals, and use `LabelProps` directly here (without wrapping it with `Partial`). This way, if there will be a required prop (from some reason), we'll only need to update `LabelProps` in a single place.